### ISSUE TITLE
Complex valued SLD

### ISSUE THREAD

**Issue by dehoni** [SasView/sasview#1954]
Naemi Leo, U Zaragoza asked how complex-valued structural SLDs are handled in the Scattering Calculator tool. If trying to load such values, SasView produces an error. This feature might be relevant, e.g. to calculate the contributions from off-resonance scattering, real anomalous scattering, the complex
anomalous scattering, and a cross term contribution for resonant X-ray scattering.
It should be tested if the complex value should be stored in a specific format to read them correctly, e.g.
Panda uses (x-yj) with the brackets and j for complex numbers.

**Comment by butlerpd** [comment]
Indeed, complex SLD values are not currently supported fully in SasView as far as I know. There was some discussion around that at several points, including the increasing importance, and I think there is a ticket (maybe even two) that discuss that.  I think @pkienzle has some thoughts on that somewhere, particularly in regard to our use of commodity GPUs (i.e. one has to be careful how it is implemented for the code to work on the GPUs ?)

**Comment by pkienzle** [comment]
Models can use complex numbers via [cl_complex.h](https://github.com/SasView/sasmodels/blob/master/sasmodels/models/lib/cl_complex.h), though the real and imaginary parts need to be sent in as separate parameters.

numba claims to support complex single and double precision, so it should be simply a matter of fixing the reader, making sure conjugates are used in the appropriate places and returning absolute values at the end in case there is some residual imaginary value after the calculation.
